runsc/cmd/gofer.go

```diff
@@ -350,6 +351,10 @@ func (g *Gofer) serve(spec *specs.Spec, conf *config.Config, root string) subcom
 		}
 		server.StartConnection(conn)
 	}
+	if unix.Getuid() == 0 {
+		log.Infof("Gofer running as root, setting the ruid and euid to %d", spec.Process.User.UID)
+		unix.Setreuid(int(spec.Process.User.UID), int(spec.Process.User.UID))
```

Comment:
If we want to handle UID this way, we probably need to do the same about GID.

Reply:
will do as well